### PR TITLE
feat: support Windows container local dumps

### PR DESCRIPTION
**Pull request by AbelHu**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
support Windows container local dumps

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: staging/cse/windows/containerdtemplate.toml [hunk 1]
```diff
@@ -46,6 +46,8 @@ state = "C:\\ProgramData\\containerd\\state"
       [plugins.cri.containerd.runtimes]
         [plugins.cri.containerd.runtimes.runhcs-wcow-process]
           runtime_type = "io.containerd.runhcs.v1"
+          container_annotations = ["io.microsoft.container.processdumplocation"], ["io.microsoft.container.processdumpcontainersubdirectory"], ["io.microsoft.wcow.processdumptype"], ["io.microsoft.wcow.processdumpcount"]
```
Comment (kiashok) [inline staging/cse/windows/containerdtemplate.toml:49]: We can update container_annotations and pod_annotations to include only the 3 following strings:

"io.microsoft.container.processdumplocation"
"io.microsoft.wcow.processdumptype"
"io.microsoft.wcow.processdumpcount"

reference: 
https://github.com/microsoft/hcsshim/blob/a27618474e0bc32622fc2890bceb77161f7e90fc/pkg/annotations/annotations.go#L233
https://github.com/microsoft/hcsshim/pull/1929

Comment (AbelHu) [reply staging/cse/windows/containerdtemplate.toml:49]: @kiashok can we add `["io.microsoft.container.processdumplocation"]` in `pod_annotations`?

File: staging/cse/windows/containerdtemplate.toml [hunk 2]
```diff
@@ -46,6 +46,8 @@ state = "C:\\ProgramData\\containerd\\state"
       [plugins.cri.containerd.runtimes]
         [plugins.cri.containerd.runtimes.runhcs-wcow-process]
           runtime_type = "io.containerd.runhcs.v1"
+          container_annotations = ["io.microsoft.container.processdumplocation"],["io.microsoft.wcow.processdumptype"],["io.microsoft.wcow.processdumpcount"]
```
Comment (kiashok) [inline staging/cse/windows/containerdtemplate.toml:49]: Could you change to the following please?
container_annotations = ["io.microsoft.container.processdumplocation", "io.microsoft.wcow.processdumptype", "io.microsoft.wcow.processdumpcount"]
pod_annotations = ["io.microsoft.container.processdumplocation", "io.microsoft.wcow.processdumptype", "io.microsoft.wcow.processdumpcount"]

https://github.com/containerd/containerd/blob/669e0786d8c390bb176068c720308d7ac93659a3/pkg/cri/config/config.go#L52 pod_annotations and container_annotations are slice of strings

Comment (AbelHu) [reply staging/cse/windows/containerdtemplate.toml]: Fixed

**Review by kiashok — APPROVED** [review]

**Comment by kiashok** [comment]
LGTM

**Review by kiashok — APPROVED** [review]
LGTM

**Review by junjiezhang1997 — APPROVED** [review]
LGTM.

**Review by ShiqianTao — APPROVED** [review]

**Pull request merged**